iohk/UpdateLogic.hs

```diff
@@ -151,12 +183,34 @@ findInstaller tempdir status = do
             Just artifacts -> do
               case head artifacts of
                 AppveyorArtifact filename "Daedalus Win64 Installer" -> do
-                  stream <- fetchUrl mempty $ T.unpack $ "https://ci.appveyor.com/api/buildjobs/" <> jobId <> "/artifacts/" <> filename
-                  print $ "appveyor URL: " <>  "https://ci.appveyor.com/api/buildjobs/" <> jobId <> "/artifacts/" <> filename
                   let
+                    url = "https://ci.appveyor.com/api/buildjobs/" <> jobId <> "/artifacts/" <> filename
                     basename = head $ drop 1 $ T.splitOn "/" filename
-                  LBS.writeFile (T.unpack $ tempdir <> "/" <> basename) stream
-                  pure $ AppveyorResult $ tempdir <> "/" <> basename
+                    outFile = tempdir <> "/" <> basename
+                  print $ "appveyor URL: " <>  url
+                  fetchCachedUrl url basename outFile
+                  pure $ AppveyorResult outFile
     other -> do
       print $ "other CI status found: " <> other
       pure Other
+
+extractBuildId :: LBS.ByteString -> BuildNumber
+extractBuildId fullLog = do
+  let
+    f1 = LBS.toStrict fullLog
+    (_, f2) = BS.breakSubstring "build id is" f1
+    f3 = BSC.takeWhile (\c -> c /= '\n') f2
+    isNumber c = (c >= '0') && (c <= '9')
+    f4 = BSC.dropWhile (\c -> (isNumber c) == False) f3
+    f5 = BSC.takeWhile isNumber f4
+  read $ BSC.unpack f5
+sampleInput :: LBS.ByteString
+sampleInput = "junk\nbuild id is 13711'\r\r\njunk"
+
+fetchTravis2 :: Repoid -> BuildNumber -> IO TravisInfo2
```

Comment:
We might try figuring out a more descriptive choice of names to disambiguate between the two..